### PR TITLE
refactor: core: Collateral test using Snapshotter

### PR DESCRIPTION
**Pull request by davidprepo**
- `# deposit` and `# withdraw` have had their setup logic refactored so it is now sequential and doesn't rely on string matching.
- decimal difference tests are grouped differently in `# deposit` and `# withdraw` in order to ease readability of `# deposit` and increase optimization of `# withdraw`

### CODE REVIEW

File: apps/smart-contracts/core/test/Collateral.test.ts [hunk 1]
```diff
@@ -808,61 +790,77 @@ describe('=> Collateral', () => {
...
-        .sub(fee)
-        .mul(parseEther('1'))
-        .div(GREATER_DECIMAL_DENOMINATOR)
-
-      await collateral.connect(sender).deposit(recipient.address, amountToDeposit)
-
-      expect(await collateral.balanceOf(recipient.address)).to.eq(recipientCTBefore.add(expectedCT))
-      expect(await collateral.balanceOf(sender.address)).to.eq(0)
+    describe("mints decimal-adjusted amount to recipient's balance", () => {
+      after(async () => {
```
Comment (davidprepo) [inline apps/smart-contracts/core/test/Collateral.test.ts:794]: After this block we must redo the setup as local variables in the test can become mismatched with the decimal back and forth.

File: apps/smart-contracts/core/test/Collateral.test.ts [hunk 2]
```diff
@@ -198,10 +188,10 @@ describe('=> Collateral', () => {
   })
 
   describe('# setManager ', () => {
+    snapshotter.usesCustomSnapshot()
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/test/Collateral.test.ts:191]:
```suggestion
    snapshotter.usesCustomSnapshot()
```
Are we not naming these anymore? i.e. `Collateral-setManager`, this is being done in 674 no?

**Review by ramenforbreakfast — APPROVED** [review]
LGTM, needs rebasing however, its using the old snapshot naming. Also question about unnamed snapshot contexts.

**Review by XavierEkkel — CHANGES_REQUESTED** [review]
please split up this PR

**Comment by ramenforbreakfast** [comment]
Closing due to deprioritization.

**Pull request closed without merging**